File: punarmilan-frontend/src/admin/services/adminVerificationService.js
import adminApi from './adminApi';

const adminVerificationService = {
    getPendingProfiles: async (params) => {
        const response = await adminApi.get('/verifications/profiles/pending', { params });
        return response.data;
    },

    approveProfile: async (id) => {
        const response = await adminApi.post(`/verifications/profiles/${id}/approve`);
        return response.data;
    },

    rejectProfile: async (id, reason) => {
        const response = await adminApi.post(`/verifications/profiles/${id}/reject`, { reason });
        return response.data;
    },

    getPendingPhotos: async (params) => {
        const response = await adminApi.get('/verifications/photos/pending', { params });
        return response.data;
    },

    approvePhotos: async (id) => {
        const response = await adminApi.post(`/verifications/photos/${id}/approve`);
        return response.data;
    },

    rejectPhotos: async (id, reason) => {
        const response = await adminApi.post(`/verifications/photos/${id}/reject`, { reason });
        return response.data;
    },

    deleteUserPhoto: async (profileId, index) => {
        const response = await adminApi.delete(`/verifications/photos/${profileId}/${index}`);
        return response.data;
    }
};

export default adminVerificationService;

File: backend/src/main/java/com/punarmilan/service/AdminVerificationService.java
package com.punarmilan.service;

import com.punarmilan.dto.ProfileDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminVerificationService {
    Page<ProfileDTO> getPendingProfiles(Pageable pageable);

    void approveProfile(Long profileId);

    void rejectProfile(Long profileId, String reason);

    Page<ProfileDTO> getPendingPhotos(Pageable pageable);

    void approvePhotos(Long profileId);

    void rejectPhotos(Long profileId, String reason);

    void deleteUserPhoto(Long profileId, Integer photoIndex);
}
